Corona-Warn-App/src/main/java/de/rki/coronawarnapp/submission/SubmissionTask.kt

```diff
@@ -0,0 +1,114 @@
+package de.rki.coronawarnapp.submission
+
+import com.google.android.gms.nearby.exposurenotification.TemporaryExposureKey
+import de.rki.coronawarnapp.appconfig.AppConfigProvider
+import de.rki.coronawarnapp.appconfig.toNewConfig
+import de.rki.coronawarnapp.playbook.Playbook
+import de.rki.coronawarnapp.server.protocols.KeyExportFormat
+import de.rki.coronawarnapp.service.submission.SubmissionService
+import de.rki.coronawarnapp.task.Task
+import de.rki.coronawarnapp.task.TaskCancellationException
+import de.rki.coronawarnapp.task.TaskFactory
+import de.rki.coronawarnapp.task.common.DefaultProgress
+import kotlinx.coroutines.channels.ConflatedBroadcastChannel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.asFlow
+import org.joda.time.Duration
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Provider
+
+class SubmissionTask @Inject constructor(
+    private val playbook: Playbook,
+    private val appConfigProvider: AppConfigProvider,
+    private val exposureKeyHistoryCalculations: ExposureKeyHistoryCalculations
+) : Task<DefaultProgress, Task.Result> {
+
+    private val internalProgress = ConflatedBroadcastChannel<DefaultProgress>()
+    override val progress: Flow<DefaultProgress> = internalProgress.asFlow()
+
+    private var isCanceled = false
+
+    override suspend fun run(arguments: Task.Arguments) = try {
+        Timber.d("Running with arguments=%s", arguments)
+        with(arguments as Arguments) {
+            playbook.submission(
+                Playbook.SubmissionData(
+                    registrationToken,
+                    getHistory(),
+                    true,
+                    supportedCountries()
+                ).also {
+                    checkCancel()
+                }
+            )
+        }
+        SubmissionService.submissionSuccessful()
```

Comment:
As a note for further refactoring, i really don't like the `SubmissionService` in its current state. Its just a agglomeration of grown over time functions with no real central goal. Id really like to spend some time here but obviously not in this PR.